manager/manager/deployer.go

```diff
@@ -83,34 +86,47 @@ func (d *deployer) GetConfiguration(cached *Configuration) (*Configuration, erro
 	return actual, nil
 }
 
-// CreateConfiguration deploys the set of resources described by a configuration.
-func (d *deployer) CreateConfiguration(configuration *Configuration) error {
+// CreateConfiguration deploys the set of resources described by a configuration and returns
+// the Configuration with status for each resource filled in.
+func (d *deployer) CreateConfiguration(configuration *Configuration) (*Configuration, error) {
 	return d.callServiceWithConfiguration("POST", "create", configuration)
 }
 
 // DeleteConfiguration deletes the set of resources described by a configuration.
-func (d *deployer) DeleteConfiguration(configuration *Configuration) error {
+func (d *deployer) DeleteConfiguration(configuration *Configuration) (*Configuration, error) {
 	return d.callServiceWithConfiguration("DELETE", "delete", configuration)
 }
 
-// PutConfiguration replaces the set of resources described by a configuration.
-func (d *deployer) PutConfiguration(configuration *Configuration) error {
+// PutConfiguration replaces the set of resources described by a configuration and returns
+// the Configuration with status for each resource filled in.
+func (d *deployer) PutConfiguration(configuration *Configuration) (*Configuration, error) {
 	return d.callServiceWithConfiguration("PUT", "replace", configuration)
 }
 
-func (d *deployer) callServiceWithConfiguration(method, operation string, configuration *Configuration) error {
+func (d *deployer) callServiceWithConfiguration(method, operation string, configuration *Configuration) (*Configuration, error) {
 	callback := func(e error) error {
 		return fmt.Errorf("cannot %s configuration: %s", operation, e)
 	}
 
 	y, err := yaml.Marshal(configuration)
 	if err != nil {
-		return callback(err)
+		return nil, callback(err)
 	}
 
 	reader := ioutil.NopCloser(bytes.NewReader(y))
-	_, err = d.callService(method, d.getBaseURL(), reader, callback)
-	return err
+	resp, err := d.callService(method, d.getBaseURL(), reader, callback)
+
+	if err != nil {
+		return nil, err
+	}
+
+	result := &Configuration{}
+	if len(resp) != 0 {
```

Comment:
Well, there's an open issue about how to handle things like delete? I think you opened a bug about it, so figured that 204 no content might apply here as well. What would you like me to do here?